spec/labor/daily_digest_email_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+
+class FakeDelegator < ActionMailer::MessageDelivery
+  # TODO: we should replace all usage of .deliver to .deliver_now
+  def deliver(*args)
+    super
+  end
+end
+
+RSpec.describe DailyDigestEmail, type: :labor do
+  let(:user) { create(:user, email_daily_digest_notifications: true) }
+  let(:author) { create(:user) }
+  let(:mock_delegator) { instance_double("FakeDelegator") }
+
+  before do
+    allow(DailyDigestMailer).to receive(:daily_digest_email) { mock_delegator }
+    allow(mock_delegator).to receive(:deliver).and_return(true)
+    user
+  end
+
+  context "when daily digest email" do
+    before { user.follow(author) }
+
+    it "send email based on highest page view count" do
+      article = create(:article, user_id: author.id, page_views_count: 50)
+      article2 = create(:article, user_id: user.id, page_views_count: 50)
```

Comment:
Nice thorough testing! Thanks for testing that the user would not receive an article they wrote and that they should only get the article with the most reads.